nav2_behavior_tree/plugins/decorator/goal_updated_controller.cpp

```diff
@@ -37,18 +36,18 @@ BT::NodeStatus GoalUpdatedController::tick()
     // Reset since we're starting a new iteration of
     // the goal updated controller (moving from IDLE to RUNNING)
 
-    BT::getInputOrBlackboard("goals", goals_);
-    BT::getInputOrBlackboard("goal", goal_);
+    goals_ = config().blackboard->get<std::vector<geometry_msgs::msg::PoseStamped>>("goals");
+    goal_ = config().blackboard->get<geometry_msgs::msg::PoseStamped>("goal");
```

Comment:
weirdly enough, here it works with `BT::getInputOrBlackboard("goal", goal_);` but I changed it to keep it consistent